enterprise/admin/utils.py

```diff
@@ -4,10 +4,102 @@
 """
 from __future__ import absolute_import, unicode_literals
 
+import unicodecsv
+from django.contrib.auth.models import User
+from django.core.exceptions import ValidationError
+from django.core.validators import validate_email
+from django.utils.translation import ugettext as _
+
+from enterprise.models import EnterpriseCustomerUser
+
 
 class UrlNames(object):
     """
     Collection on URL names used in admin
     """
     URL_PREFIX = "enterprise_"
     MANAGE_LEARNERS = URL_PREFIX + "manage_learners"
+
+
+class ValidationMessages(object):
+    """
+    Namespace class for validation messages.
+    """
+    INVALID_EMAIL_OR_USERNAME = _("{email_or_username} does not appear to be a valid email or known username")
+    USER_ALREADY_REGISTERED = _("User with email {email} is already registered with Enterprise Customer {ec_name}")
+    MISSING_EXPECRED_COLUMNS = _(
+        "Row {row_number} expected to have [{expected_columns}] columns, but had [{actual_columns}]"
+    )
+
+    NO_FIELDS_SPECIFIED = _("Either \"Email or Username\" or \"CSV bulk upload\" must be specified, neither were")
+    BOTH_FIELDS_SPECIFIED = _("Either \"Email or Username\" or \"CSV bulk upload\" must be specified, both were.")
+
+
+def parse_csv(file_stream, expected_columns=None):
+    """
+    Parse csv file and return a stream of dictionaries representing each row.
+
+    First line of CSV file must contain column headers.
+
+    Arguments:
+         file_stream: input file
+         expected_columns (set[unicode]): columns that are expected to be present
+
+    Yields:
+        dict: CSV line parsed into a dictionary.
+    """
+    reader = unicodecsv.DictReader(file_stream, encoding="utf-8")
+
+    # "yield from reader" would be nicer, but we're on python2.7 yet.
+    for idx, row in enumerate(reader):
+        if expected_columns:
+            if not set(row.keys()) >= expected_columns:
+                raise ValidationError(ValidationMessages.MISSING_EXPECRED_COLUMNS.format(
+                    row_number=idx, expected_columns=", ".join(expected_columns), actual_columns=", ".join(row.keys())
+                ))
+        yield row
+
+
+def username_or_email__to__email(email_or_username):
+    """
+    Convert email_or_username to email.
+
+    Returns:
+        str: If `email_or_username` was a username returns user's email, otherwise assumes it was an email and returns
+             as is.
+    """
+    try:
+        user = User.objects.get(username=email_or_username)
```

Comment:
@mattdrayer Email addresses are [not valid usernames on edx-platform](https://github.com/edx/edx-platform/blob/master/common/djangoapps/student/forms.py#L123-L124).  Since this requirement is enforced by the registration form rather than the model, it's certainly possible to circumvent it using managment commands, but I agree with your assessment that this corner case does not matter. :)